File: src/Samples/CSharpApp/App.ViewModel/MainWindowViewModel.cs
namespace SampleApp.ViewModelCS
{
    using System.Collections.ObjectModel;
    using System.Linq;
    using Utopia.ViewModel;
    using SampleApp.Model;

    public class MainWindowViewModel : ViewModelBase
    {
        PageViewModel _currentPage;

        public PageViewModel CurrentPage
        {
            get { return _currentPage; }
            set
            {
                _currentPage = value;
                OnPropertyChanged("CurrentPage");
            }
        }

        public ObservableCollection<PageViewModel> PageViewModels { get; private set; }

        public MainWindowViewModel() : base()
        {
            var calcEngine = new CalcEngineModel();

            PageViewModels = new ObservableCollection<PageViewModel>();
            PageViewModels.Add(new GraphViewModel(calcEngine));

            CurrentPage = PageViewModels.FirstOrDefault();
        }

        public string Title {get { return "Calc FrameWork v0.1"; } }
    }
}

File: src/Samples/FSharpApp/App.ViewModel/MainWindowViewModel.fs
﻿namespace FSharpApp.ViewModel

open System.Collections.ObjectModel
open Utopia.ViewModel
open App.Model

type MainWindowViewModel() =
    inherit ViewModelBase()
    let mutable currentpage:PageViewModel = new PageViewModel()
    let pages = new ObservableCollection<PageViewModel>()

    let ce = CalcEngineModel()

    // viewmodels of app
    let graphVM = new GraphViewModel(ce)

    let addPage page = pages.Add(page :> PageViewModel)

    do
          graphVM |> addPage
          currentpage <- pages.[0]
   
    member this.Title = "Calc FrameWork v0.1"
    member this.PageViewModels = pages